server/src/controllers/userController.js

```diff
@@ -0,0 +1,29 @@
+
+import User from '../database/models/user';
+
+const createNewUser = (req, res) => {
+  const { email, username, password } = req.body;
+
+  const newUser = new User({
+    email,
+    username,
+    password,
+  });
+  return newUser
+    .save()
+    .then(() => {
```

Comment:
Why not async/await?

Reply:
I used .then and .catch just to have a Promise based workflow. I'd like to keep it like that, unless you have a reason for going with async/await?

Reply:
Check [this](https://hackernoon.com/6-reasons-why-javascripts-async-await-blows-promises-away-tutorial-c7ec10518dd9) for async await.
Also its a new feature in ES so you must upgrade your codebase to latest specs :)

Reply:
Okay, I will update this. Thank you for the suggestion.